leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -189,90 +173,50 @@ private PaxosQuorumChecker() {
             }
         }
 
-        List<Throwable> toLog = Lists.newArrayList();
+        List<Throwable> encounteredErrors = Lists.newArrayList();
         boolean interrupted = false;
-        List<RESPONSE> receivedResponses = new ArrayList<RESPONSE>();
-        int acksRecieved = 0;
-        int nacksRecieved = 0;
-
+        PaxosResponses<RESPONSE> receivedResponses = new PaxosResponses<>(remotes.size(), quorumSize,
+                shortcircuitIfQuorumImpossible);
         try {
-            long deadline = System.nanoTime() + TimeUnit.SECONDS.toNanos(remoteRequestTimeoutInSec);
-            // handle responses
-            while (acksRecieved < quorumSize) {
+            long deadline = System.nanoTime() + remoteRequestTimeout.toNanos();
+            while (receivedResponses.shouldProcessNextRequest()) {
                 try {
-                    // check if quorum is impossible (nack quorum failure)
-                    if (shortcircuitIfQuorumImpossible && nacksRecieved > remotes.size() - quorumSize) {
```

Comment:
Why didn't this get us out of this logic before? Were we indefinitely retrying deeper down?

Reply:
`shortcircuitIfQuorumImpossible` was `false` and setting it to `true` isn't what we want

Reply:
and yes, we were retrying indefinitely

Reply:
Oops, yeah, this is a `&&`